app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -70,25 +88,51 @@ class AgendaRepositoryImpl(
             }
         }
 
-        return db.getReminderDao().getRemindersOfDateFlow(startEpochMilli, endEpochMilli).map {
+        val reminders = db.getReminderDao().getRemindersOfDateFlow(
+            startEpochMilli = startEpochMilli,
+            endEpochMilli = endEpochMilli
+        ).map {
             it.map { reminderEntity ->
                 reminderEntity.toReminder()
             }.sortedBy { reminder ->
                 reminder.startDateAndTime
             }
         }
-        // Todo return task and event agenda items
+        val tasks = db.getTaskDao().getTasksOfDateFlow(
+            startEpochMilli = startEpochMilli,
+            endEpochMilli = endEpochMilli
+        ).map {
+            it.map { taskEntity ->
+                taskEntity.toTask()
+            }.sortedBy { task ->
+                task.startDateAndTime
+            }
+        }
+        return reminders.combine(tasks) { reminderFlow, taskFlow ->
+            reminderFlow + taskFlow
+        }.map {
+            it.sortedBy { agendaItem ->
+                agendaItem.startDateAndTime
+            }
+        }
```

Comment:
The `sortedBy` on top seems redundant to me since you sort it again after combining them